### PR TITLE
Feature/basic email reminder settings

### PR DESCRIPTION
**Pull request by ulferts**
Adds the menu item "Email reminders" to the my account and allows to specify the times at which the non read notifications are to be sent as a digest/reminder email.

image

**NOTE** The "Immediately when someone @mentions me" is out of scope for this PR

### TODO
* [x] Have a menu item "Email reminders" that triggers an angular page component
* [x] Have an input component that allows managing the execution time as specified in the figma.
* [x] Extend the API & Model & Contract to allow saving the newly added items (included in #9619 which this PR bases on)
* [x] Reminder functionality is active and configured to run at "08:00" by default
* ~Depending on whether #9611 is merged into dev or not, use the added settings to schedule the reminder sending background jobs.~ 
  * ~replace https://github.com/opf/openproject/blob/dev/spec/features/notifications/digest_mail_spec.rb~

https://community.openproject.org/wp/38707

### CODE REVIEW

File: app/controllers/my_controller.rb [hunk 1]
```diff
@@ -81,23 +82,30 @@ def change_password
...
-      flash[:info] = [
-        t('my.access_token.notice_reset_token', type: 'RSS').html_safe,
-        content_tag(:strong, token.plain_value),
-        t('my.access_token.token_value_warning')
-      ]
-    end
+    token = Token::RSS.create!(user: current_user)
+    flash[:info] = [
+      # rubocop:disable Rails/OutputSafety
+      t('my.access_token.notice_reset_token', type: 'RSS').html_safe,
```
Comment (oliverguenther) [inline app/controllers/my_controller.rb:104]: We can avoid calling `.html_safe` by chaniging the i18n key to something with a `*_html` suffix. Not meant as a review feedback for this PR, just as a reminder for us.

File: db/migrate/20210902201126_cleanup_user_preferences.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+class CleanupUserPreferences < ActiveRecord::Migration[6.1]
```
Comment (oliverguenther) [inline db/migrate/20210902201126_cleanup_user_preferences.rb:1]: Thanks! 🙇

File: frontend/src/app/features/user-preferences/reminder-settings/reminder-time/reminder-settings-daily-time.component.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  // Whether the reminder are active at all.
+  public enabled$ = this.query.dailyRemindersEnabled$;
+
+  // The active times as present in the store interleaved with the inactive
+  // times.
+  public selectedTimes$ = this
+    .query
+    .dailyRemindersTimes$
+    .pipe(map((times) => {
+      const activeTimes = Array(...times);
```
Comment (oliverguenther) [inline frontend/src/app/features/user-preferences/reminder-settings/reminder-time/reminder-settings-daily-time.component.ts:44]: Side note: I prefer the style `[...times]` but I'm not aware of any semantical differences

File: frontend/src/app/features/user-preferences/reminder-settings/reminder-time/reminder-settings-daily-time.component.ts [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+          activeTimes.splice(inactiveTime.position, 0, inactiveTime.time);
+        });
+
+      return activeTimes;
+    }));
+
+  // Times that are truly active:
+  // * the reminders are not disabled completely
+  // * the times are not inactive individually.
+  public activeTimes$ = combineLatest(
```
Comment (oliverguenther) [inline frontend/src/app/features/user-preferences/reminder-settings/reminder-time/reminder-settings-daily-time.component.ts:59]: combineLatest without array syntax is deprecated.

Use `combineLatest([obs1, obs2])...` instead

File: modules/backlogs/lib/open_project/backlogs/engine.rb [hunk 5]
```diff
@@ -131,6 +131,20 @@ def self.settings
       next if Versions::BaseContract.included_modules.include?(OpenProject::Backlogs::Patches::Versions::BaseContractPatch)
 
       Versions::BaseContract.prepend(OpenProject::Backlogs::Patches::Versions::BaseContractPatch)
+
+      # Add available settings to the user preferences
+      UserPreferences::Schema.merge!(
```
Comment (oliverguenther) [inline modules/backlogs/lib/open_project/backlogs/engine.rb:136]: 👍

**Review by oliverguenther — APPROVED** [review]
Works nicely and code is as expected. I added some nitpick comments that are not in need of fixing.

File: frontend/src/app/features/user-preferences/reminder-settings/reminder-time/reminder-settings-daily-time.component.ts (hunk 4 above)
Comment (ulferts) [reply frontend/src/app/features/user-preferences/reminder-settings/reminder-time/reminder-settings-daily-time.component.ts:59]: Ah, didn't know that. Thanks for pointing it out.

**Pull request merged**